tests/runtimes/test_runtime_handler.py

```diff
@@ -0,0 +1,690 @@
+        resources,
+        expected_crds=expected_crds,
+        expected_pods=expected_pods,
+        expected_services=expected_services,
+    )
+
+
+def _assert_list_resources_response(
+    resources, expected_crds=None, expected_pods=None, expected_services=None
+):
+    if expected_crds is None:
+        expected_crds = []
+    else:
+        expected_crds = expected_crds["items"]
+    if expected_pods is None:
+        expected_pods = []
+    if expected_services is None:
+        expected_services = []
+    else:
+        expected_services = expected_services.items
+    assert len(resources["crd_resources"]) == len(expected_crds)
+    for index, crd in enumerate(expected_crds):
+        assert resources["crd_resources"][index]["name"] == crd["metadata"]["name"]
+        assert resources["crd_resources"][index]["labels"] == crd["metadata"]["labels"]
+        assert resources["crd_resources"][index]["status"] == crd["status"]
+    assert len(resources["pod_resources"]) == len(expected_pods)
+    for index, pod in enumerate(expected_pods):
+        pod_dict = pod.to_dict()
+        assert resources["pod_resources"][index]["name"] == pod_dict["metadata"]["name"]
+        assert (
+            resources["pod_resources"][index]["labels"]
+            == pod_dict["metadata"]["labels"]
+        )
+        assert resources["pod_resources"][index]["status"] == pod_dict["status"]
+    if expected_services:
+        assert len(resources["service_resources"]) == len(expected_services)
+        for index, service in enumerate(expected_services):
+            assert (
+                resources["service_resources"][index]["name"] == service.metadata.name
+            )
+            assert (
+                resources["service_resources"][index]["labels"]
+                == service.metadata.labels
+            )
+
+
+def _mock_list_pods(k8s_helper_mock, pod_dicts):
+    pod_mocks = []
+    for pod_dict in pod_dicts:
+        pod_mock = unittest.mock.Mock()
+        pod_mock.to_dict.return_value = pod_dict
+        pod_mocks.append(pod_mock)
+    k8s_helper_mock.list_pods.return_value = pod_mocks
+    return pod_mocks
+
+
+def _mock_list_crds(k8s_helper_mock, crd_dicts):
+    crds = {
+        "items": crd_dicts,
+    }
+    k8s_helper_mock.crdapi.list_namespaced_custom_object.return_value = crds
+    return crds
+
+
+def _mock_list_services(k8s_helper_mock, service_dicts):
+    service_mocks = []
+    for service_dict in service_dicts:
+        service_mock = unittest.mock.Mock()
+        service_mock.metadata.name.return_value = service_dict["metadata"]["name"]
+        service_mock.metadata.labels.return_value = service_dict["metadata"]["labels"]
+        service_mocks.append(service_mock)
+    services_mock = unittest.mock.Mock()
+    services_mock.items = service_mocks
+    k8s_helper_mock.v1api.list_namespaced_service.return_value = services_mock
+    return services_mock
+
+
+def _mock_list_kubejob_pods(k8s_helper_mock):
```

Comment:
All the mock functions below are holding enormous dictionaries with hardcoded data, which makes the test file a bit unreadable.
As all these are completely static, how about keeping them in json/yaml files in an assets directory, and then these mocks can just open the file, parse the json/yaml and add it to the mock.
I think it will be a lot cleaner.
WDYT?

Reply:
Currently you're right, it will be a lot cleaner, but I'm expecting this dictionaries to become dynamic when I'll add the delete resources tests, so leaving as is for now